xmantissa/port.py

```diff
@@ -437,79 +499,19 @@ def postOptions(self):
                     raise SystemExit(1)
-                except KeyError:
-                    print "Unrecognized port type."
-                    raise SystemExit(1)
-                except SSL.Error, e:
-                    if self._pemFormatError in e.args[0]:
-                        print 'Certificate file must use PEM format.'
-                        raise SystemExit(1)
-                    elif self._noSuchFileError in e.args[0]:
-                        if self._certFileError in e.args[0]:
-                            print "Specified certificate file does not exist."
-                            raise SystemExit(1)
-                        elif self._keyFileError in e.args[0]:
-                            print "Specified private key file does not exist."
-                            raise SystemExit(1)
-                        else:
-                            # Note, no test coverage.
-                            raise
-                    else:
-                        # Note, no test coverage.
-                        raise
-                else:
-                    try:
-                        method = getattr(self, 'create_' + kind)
-                    except AttributeError:
-                        print "Unsupported port type."
-                        raise SystemExit(1)
-                    else:
-                        port = method(store, *args, **kwargs)
-                        installOn(port, store)
-                        print "Created."
+                port = StringEndpointPort(
+                    store=store, description=description, factory=factory)
+                installOn(port, store)
+                print "Created."
         raise SystemExit(0)
 
 
-    def create_TCP(self, store, port, factory, backlog, interface):
-        """
-        Create a new L{TCPPort} with the specified parameters.
-        """
-        return TCPPort(
-            store=store, portNumber=port,
-            factory=factory, interface=self.decodeCommandLine(interface))
-
-
-    def create_SSL(self, store, port, factory, context, backlog, interface):
-        """
-        Create a new L{SSLPort} with the specified parameters.
-
-        @type context: L{DefaultOpenSSLContextFactory}
-        """
-        key = context.privateKeyFileName
-        cert = context.certificateFileName
-        if key != cert:
-            print "You must specify the same file for certKey and privateKey."
-            raise SystemExit(1)
-        elif context.sslmethod != SSL.SSLv23_METHOD:
-            print "Only SSLv23_METHOD is supported."
-            raise SystemExit(1)
-        else:
-            port = SSLPort(
-                store=store, portNumber=port, factory=factory,
-                interface=self.decodeCommandLine(interface))
-            targetDir = store.filesdir.child(str(port.storeID))
-            targetDir.makedirs()
-            targetPath = targetDir.child("cert.pem")
```

Comment:
Worth noting that this is a change in behavior: previously, the certificate would be copied into the store, now, it won't be.  This is the first piece of external configuration that an axiomatic store would need to run.

OTOH, I have *no* idea how you're supposed to do cert renewals for mantissa, so maybe this is better :-).